crates/model/src/defi/dex.rs

```diff
@@ -96,3 +99,46 @@ impl Display for Dex {
         write!(f, "Dex(chain={}, name={})", self.chain, self.name)
     }
 }
+
+impl From<Pool> for CurrencyPair {
+    fn from(p: Pool) -> Self {
+        let symbol = Symbol::from(format!("{}/{}", p.token0.symbol, p.token1.symbol));
+        let id = InstrumentId::new(symbol, Venue::from("DEFI"));
+
+        let size_precision = p.token0.decimals;
+        let price_precision = p.token0.decimals.saturating_add(p.token1.decimals);
```

Comment:
This should probably be the `min(token1.decimals, FIXED_PRECISION)` (because we're currently limited to 16 rather than the full 18 which much of DeFi will need).